bot/exts/recruitment/talentpool/_review.py

```diff
@@ -0,0 +1,316 @@
+
+from dateutil.parser import isoparse
+from dateutil.relativedelta import relativedelta
+from discord import Emoji, Member, Message, TextChannel
+from discord.ext.commands import Context
+
+from bot.api import ResponseCodeError
+from bot.bot import Bot
+from bot.constants import Channels, Guild, Roles
+from bot.utils.scheduling import Scheduler
+from bot.utils.time import get_time_delta, humanize_delta, time_since
+
+if typing.TYPE_CHECKING:
+    from bot.exts.recruitment.talentpool._cog import TalentPool
+
+log = logging.getLogger(__name__)
+
+# Maximum amount of days before an automatic review is posted.
+MAX_DAYS_IN_POOL = 30
+
+# Maximum amount of characters allowed in a message
+MAX_MESSAGE_SIZE = 2000
+
+
+class Reviewer:
+    """Schedules, formats, and publishes reviews of helper nominees."""
+
+    def __init__(self, name: str, bot: Bot, pool: 'TalentPool'):
+        self.bot = bot
+        self._pool = pool
+        self._review_scheduler = Scheduler(name)
+
+    def __contains__(self, user_id: int) -> bool:
+        """Return True if the user with ID user_id is scheduled for review, False otherwise."""
+        return user_id in self._review_scheduler
+
+    async def reschedule_reviews(self) -> None:
+        """Reschedule all active nominations to be reviewed at the appropriate time."""
+        log.trace("Rescheduling reviews")
+        await self.bot.wait_until_guild_available()
+        # TODO Once the watch channel is removed, this can be done in a smarter way, e.g create a sync function.
+        await self._pool.fetch_user_cache()
+
+        for user_id, user_data in self._pool.watched_users.items():
+            if not user_data["reviewed"]:
+                self.schedule_review(user_id)
+
+    def schedule_review(self, user_id: int) -> None:
+        """Schedules a single user for review."""
+        log.trace(f"Scheduling review of user with ID {user_id}")
+
+        user_data = self._pool.watched_users[user_id]
+        inserted_at = isoparse(user_data['inserted_at']).replace(tzinfo=None)
+        review_at = inserted_at + timedelta(days=MAX_DAYS_IN_POOL)
+
+        self._review_scheduler.schedule_at(review_at, user_id, self.post_review(user_id, update_database=True))
+
+    async def post_review(self, user_id: int, update_database: bool) -> None:
+        """Format a generic review of a user and post it to the mod announcements channel."""
+        log.trace(f"Posting the review of {user_id}")
+
+        nomination = self._pool.watched_users[user_id]
+        if not nomination:
+            log.trace(f"There doesn't appear to be an active nomination for {user_id}")
+            return
+
+        guild = self.bot.get_guild(Guild.id)
+        channel = guild.get_channel(Channels.mod_announcements)
+        member = guild.get_member(user_id)
+        if not member:
+            channel.send(f"I tried to review the user with ID `{user_id}`, but they don't appear to be on the server 😔")
```

Comment:
If this is exited before marking them as reviewed, wouldn't this message be sent whenever the bot restarts? It probably wouldn't be too much of a problem, but I think it might make sense to just unnominate them if they come up for review and aren't in the server (along with this message).

Reply:
Should it unnominate, or just mark as reviewed first?

Reply:
Hmm, both would be fine, as this doesn't happen too often. Marking as reviewed may be better though, as it would allow more context to be added to the unnomination if necessary, so yeah, mark as reviewed sounds good.